plugins/inputs/system/ps.go

```diff
@@ -73,53 +94,38 @@ func (s *SystemPS) DiskUsage(
 		return nil, nil, err
 	}
 
-	// Make a "set" out of the filter slice
-	mountPointFilterSet := make(map[string]bool)
+	mountPointFilterSet := newSet()
 	for _, filter := range mountPointFilter {
-		mountPointFilterSet[filter] = true
+		mountPointFilterSet.add(filter)
 	}
-	fstypeExcludeSet := make(map[string]bool)
+	fstypeExcludeSet := newSet()
 	for _, filter := range fstypeExclude {
-		fstypeExcludeSet[filter] = true
+		fstypeExcludeSet.add(filter)
 	}
-	paths := make(map[string]bool)
+	paths := newSet()
 	for _, part := range parts {
-		paths[part.Mountpoint] = true
+		paths.add(part.Mountpoint)
 	}
 
 	// Autofs mounts indicate a potential mount, the partition will also be
 	// listed with the actual filesystem when mounted.  Ignore the autofs
 	// partition to avoid triggering a mount.
-	fstypeExcludeSet["autofs"] = true
+	fstypeExcludeSet.add("autofs")
 
 	var usage []*disk.UsageStat
 	var partitions []*disk.PartitionStat
 	hostMountPrefix := s.OSGetenv("HOST_MOUNT_PREFIX")
 
-	for i := range parts {
-		p := parts[i]
-
-		if s.Log != nil {
-			s.Log.Debugf("[SystemPS] partition %d: %v", i, p)
-		}
-
-		if len(mountPointFilter) > 0 {
-			// If the mount point is not a member of the filter set,
-			// don't gather info on it.
-			if _, ok := mountPointFilterSet[p.Mountpoint]; !ok {
-				if s.Log != nil {
-					s.Log.Debug("[SystemPS] => dropped by mount-point filter")
-				}
-				continue
-			}
+	for _, p := range parts {
+		// If the mount point is not a member of the filter set,
+		// don't gather info on it.
+		if !mountPointFilterSet.has(p.Mountpoint) {
+			continue
 		}
 
 		// If the mount point is a member of the exclude set,
 		// don't gather info on it.
-		if _, ok := fstypeExcludeSet[p.Fstype]; ok {
-			if s.Log != nil {
-				s.Log.Debug("[SystemPS] => dropped by filesystem-type filter")
-			}
+		if fstypeExcludeSet.has(p.Fstype) {
 			continue
 		}
```

Comment:
How about we remove the mount-point from the set as it is likely never changing the FS-type?

Reply:
In this PR I only addressed the verbose logging and didn't want to change behavior so I didn't think through any improvements (other than adding the set type).

I'm not sure I follow you about removing the mount point. Each gather we get disk usage from gopsutil. It returns usage by filesystem/partition/mountpoint. In this func we skip the filesystem if it's a blacklisted fs type. If we remove it, it will be back in the next gather because gopsutil returns it every time we get disk usage.

Reply:
Understood and I agree here.